akka-remote/src/main/scala/akka/remote/Remoting.scala

```diff
@@ -548,18 +555,28 @@ private[remote] class EndpointManager(conf: Config, log: LoggingAdapter) extends
             "address cannot be quarantined without knowing the UID, gating instead for {} ms.",
             address, settings.RetryGateClosedFor.toMillis)
           endpoints.markAsFailed(endpoint, Deadline.now + settings.RetryGateClosedFor)
-        case (Some(Pass(endpoint, Some(currentUid), _)), Some(quarantineUid)) if currentUid == quarantineUid ⇒
-          context.stop(endpoint)
-        case _ ⇒
-        // Do nothing, because either:
-        // A: we don't know yet the UID of the writer, it will be checked against current quarantine state later
-        // B: we know the UID, but it does not match with the UID to be quarantined
+        case (Some(Pass(endpoint, uidOption, refuseUidOption)), Some(quarantineUid)) ⇒
+          uidOption match {
+            case Some(uid) if uid == quarantineUid ⇒
```

Comment:
It's more concise and functionally equivalent. You spare one variable and instead match directly against the existing one:
```scala
case Some(`quarantineUid`) ⇒
```